File: backend/database/elements.py
from sqlalchemy import Column, Integer, String, Float
from .base import BaseModel

class ElementModel(BaseModel):
    __tablename__ = "elements"

    atomic_number = Column(Integer, unique=True, index=True)
    symbol = Column(String, unique=True, index=True)
    name = Column(String, unique=True)
    group = Column(String)
    period = Column(Integer)
    block = Column(String)
    atomic_weight = Column(Float)
    density = Column(Float, nullable=True)
    melting_point = Column(Float, nullable=True)
    boiling_point = Column(Float, nullable=True)
    specific_heat = Column(Float, nullable=True)
    electronegativity = Column(Float, nullable=True)
    abundance = Column(Float, nullable=True)
    origin = Column(String)
    phase = Column(String)
    

File: frontend/src/lib/types/elements.ts
export interface ElementType {
    id: number;
    name: string;

    atomic_number: number;
    atomic_weight: number;
    symbol: string;
    group: number;
    period: number;
    block: string;
    origin: string;
    phase: string;

    density?: number;
    melting_point?: number;
    boiling_point?: number;
    specific_heat?: number;
    electronegativity?: number;
    abundance?: number;
    // category: string;

    created_at?: string;
    updated_at?: string;
}